papas/path.py

```diff
@@ -114,11 +126,78 @@ def distquad (time):
         minim_answer = opti.bracket(distquad, xa = -0.5e-14, xb = 0.5e-14)
         self.closest_t = minim_answer[1]
         vector_IP = self.point_at_time(minim_answer[1]) - vertex
-        self.signIP= vector_IP.Dot(self.p4.Vect().Unit())
-        self.IP = minim_answer[4]**(1.0/2)*sign(self.signIP)
-        #self.closest=TVector3(self.fx(t), self.fy(t), self.fz(t))
+        Pj = jet.p4().Vect().Unit()
+        signIP  = vector_IP.Dot(Pj)
+        self.IP = minim_answer[4]**(1.0/2)*sign(signIP)
+        
+        x,y,z = self.coord_at_time(minim_answer[1])
+        self.IPx, self.IPy, self.IPz = (x-vertex.x()),  (y-vertex.y()), (z-vertex.z())
+       
+    def compute_theta_0(self):
```

Comment:
maybe this function should take x/xo as a parameter. 
again, you don't know the detector here